shell/platform/linux/fl_platform_channel.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_platform_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_method_codec.h"
+
+struct _FlPlatformChannel {
+  FlMethodChannel parent_instance;
+};
+
+G_DEFINE_TYPE(FlPlatformChannel,
+              fl_platform_channel,
+              fl_method_channel_get_type())
+
+static void method_call_cb(FlMethodChannel* channel,
+                           const gchar* method,
+                           FlValue* args,
+                           FlMethodChannelResponseHandle* response_handle,
+                           gpointer user_data) {
+  // FlPlatformChannel* self = FL_PLATFORM_CHANNEL(user_data);
+
+  if (strcmp(method, "Clipboard.setData") == 0) {
+    g_debug("Got Clipboard.setData\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "Clipboard.getData") == 0) {
+    g_debug("Got Clipboard.getData\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "HapticFeedback.vibrate") == 0) {
```

Comment:
I wouldn't include cases for things we'll likely never implement on desktop.

Reply:
You say that, but GTK does work on various devices (phones, tablets, 2 in one laptops) and so is likely to have hooks for things like this. I don't recall any haptic API but I expect one will exist in the future if not now.